pkg/build/builder/util.go

```diff
@@ -201,3 +173,39 @@ func addBuildLabels(labels map[string]string, build *buildapi.Build) {
 	labels[buildapi.DefaultDockerLabelNamespace+"build.name"] = build.Name
 	labels[buildapi.DefaultDockerLabelNamespace+"build.namespace"] = build.Namespace
 }
+
+// getCgroupParent determines the parent cgroup for a container from
+// within that container.
+func getCgroupParent() (string, error) {
+	cgMap, err := cgroups.ParseCgroupFile("/proc/self/cgroup")
```

Comment:
well it will only run inside a linux container so that's ok but i'm not sure what we do about the cross-compile build process.  I assume this isn't the first instance of something like this.